cmd/gateway/commands_test.go

```diff
@@ -676,3 +662,50 @@ func TestGetBuildInfo(t *testing.T) {
 	g.Expect(commitTime).To(Not(Equal("unknown")))
 	g.Expect(dirtyBuild).To(Not(Equal("unknown")))
 }
+
+func TestCreateGatewayPodConfig(t *testing.T) {
+	t.Parallel()
+	g := NewWithT(t)
+
+	// Order matters here
+	// We start with all env vars set
+	g.Expect(os.Setenv("POD_IP", "10.0.0.0")).To(Succeed())
+	g.Expect(os.Setenv("POD_UID", "1234")).To(Succeed())
+	g.Expect(os.Setenv("POD_NAMESPACE", "default")).To(Succeed())
+	g.Expect(os.Setenv("POD_NAME", "my-pod")).To(Succeed())
+
+	expCfg := config.GatewayPodConfig{
+		PodIP:       "10.0.0.0",
+		ServiceName: "svc",
+		Namespace:   "default",
+		Name:        "my-pod",
+		UID:         "1234",
+	}
+	cfg, err := createGatewayPodConfig("svc")
+	g.Expect(err).To(Not(HaveOccurred()))
+	g.Expect(cfg).To(Equal(expCfg))
+
+	// unset name
+	g.Expect(os.Unsetenv("POD_NAME")).To(Succeed())
+	cfg, err = createGatewayPodConfig("svc")
+	g.Expect(err).To(MatchError(errors.New("environment variable POD_NAME not set")))
+	g.Expect(cfg).To(Equal(config.GatewayPodConfig{}))
+
+	// unset namespace
+	g.Expect(os.Unsetenv("POD_NAMESPACE")).To(Succeed())
+	cfg, err = createGatewayPodConfig("svc")
+	g.Expect(err).To(MatchError(errors.New("environment variable POD_NAMESPACE not set")))
+	g.Expect(cfg).To(Equal(config.GatewayPodConfig{}))
+
+	// unset UUID
```

Comment:
I think this is what you meant? maybe change it to POD_UID

```suggestion
	// unset UID
```